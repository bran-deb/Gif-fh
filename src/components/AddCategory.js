import React, { useState } from 'react'
import PropTypes from 'prop-types'

const AddCategory = ({ setCategories }) => {

    const [inputValue, setInputValue] = useState('')

    const handleInputChange = (e) => {
        setInputValue(e.target.value)
    }
    //Enter
    const handleSubmit = (e) => {
        e.preventDefault()
        //validacion
        if (inputValue.trim().length > 2) {
            //el callback guarda el estado anterior (categories)
            setCategories(cats => [...cats, inputValue])
            setInputValue('')
        }
    }

    return (
        <form onSubmit={handleSubmit}>
            <input
                type='text'
                value={inputValue}
                onChange={handleInputChange}
            />
        </form>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}

export default AddCategory
