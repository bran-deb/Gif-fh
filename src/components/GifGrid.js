import React from 'react'
import { useState, useEffect } from 'react'

export const GifGrid = ({ category }) => {
    //hacemos que el componente se ejecute una sola vez
    useEffect(() => {
        getGifs()
    }, [])

    const [count, setCount] = useState(0)

    const getGifs = async () => {
        const url = 'https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&api_key=i6WreDgXbyKNG06qdfOz08YlRDqCbtUx'
        const resp = await fetch(url)
        const { data } = await resp.json()

        const gifs = data.map(img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url,
            }
        })
        console.log(gifs)
    }

    return (
        <div>
            <h3>{category}</h3>
            <h3>{count}</h3>
            <button onClick={() => setCount(count + 1)}>+1</button>
        </div>
    )
}