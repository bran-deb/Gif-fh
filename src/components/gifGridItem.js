import React from 'react'

export const GifGridItem = ({ title, url }) => {

    console.log(title, url)

    return (
        <div>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}
