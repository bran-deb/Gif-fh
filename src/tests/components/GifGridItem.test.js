import React from 'react';
import { shallow } from 'enzyme'
import { GifGridItem } from '../../components/gifGridItem'

describe('Pruebas en <GidGridItem/>', () => {

    const title = 'Un titulo'
    const url = 'https://un-url.jpg'

    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    //test proptypes y snapshot
    test('debe de mostrar el componente correctamente', () => {

        expect(wrapper).toMatchSnapshot()
    })

    test('el parrafo <p/> debe de tener el title', () => {

        //seleccionamos el parrafo
        const p = wrapper.find('p')
        //comparamos el texto del parrafo sin espacios con el title
        expect(p.text().trim()).toBe(title)
    })

    test('el <img/> debe de tener una img con el src y el alt', () => {

        //seleccionamos la img
        const img = wrapper.find('img')
        //vemos todos los atributos de img con .props() o con .html()
        // console.log(img.props())

        //solo mostramos la prop de src de dos formas
        // console.log(img.props().src)
        // console.log(img.prop('src'))

        expect(img.prop('src')).toBe(url)
        expect(img.prop('alt')).toBe(title)
    })

    test('el <div/> debe de tener la clase animate__fadeIn', () => {
        //seleccionamos div
        const div = wrapper.find('div')
        // console.log(div.prop('className').includes('animate__fadeIn'))
        // console.log(div.hasClass('animate__fadeIn'))

        //mostrar las clases con estos dos metodos son equivalentes
        expect(div.prop('className').includes('animate__fadeIn')).toBe(true)
        expect(div.hasClass('animate__fadeIn')).toBe(true)
    })

})
