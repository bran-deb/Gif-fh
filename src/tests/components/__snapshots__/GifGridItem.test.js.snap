// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GidGridItem/> debe de mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://un-url.jpg"
  />
  <p>
    Un titulo
  </p>
</div>
`;
